src/Extensions/Hosting.Services.Web/Middlewares/UserIdentity/UserIdentityMiddleware.cs

```diff
@@ -48,7 +48,8 @@ async Task IMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
 
 		internal async Task<string> CreateUserHashAsync(HttpContext context)
 		{
-			using IMemoryOwner<byte> uidMemoryOwner = MemoryPool<byte>.Shared.Rent(m_maxIdentitySize + m_saltProvider.GetSalt().Length);
+			int len = m_saltProvider.GetSalt().Length;
```

Comment:
I've noticed that we are getting Salt too many times in this method (not the fault of this PR). Could you please try to change it to get salt once and reuse it?